sdk/python/kfp/components/importer_node.py

```diff
@@ -56,18 +58,65 @@ def pipeline_with_importer():
           train(dataset=importer1.output)
     """
 
+    component_inputs: Dict[str, structures.InputSpec] = {}
+    call_inputs: Dict[str, Any] = {}
+
+    def traverse_dict_and_create_metadata_inputs(d: Any) -> Any:
+        if isinstance(d, pipeline_channel.PipelineParameterChannel):
+            reversed_call_inputs = {
+                pipeline_param_chan: name
+                for name, pipeline_param_chan in call_inputs.items()
+            }
+
+            # minimizes importer spec interface by not creating new
+            # inputspec/parameters if the same input is used multiple places
+            # in metadata
+            unique_name = reversed_call_inputs.get(
+                d,
+                make_placeholder_unique(
+                    METADATA_KEY,
+                    list(call_inputs),
+                    '-',
+                ),
+            )
+
+            call_inputs[unique_name] = d
+            component_inputs[unique_name] = structures.InputSpec(
+                type=d.channel_type)
+
+            return placeholders.InputValuePlaceholder(
+                input_name=unique_name).to_placeholder_string()
+
+        elif isinstance(d, dict):
+            return {
+                traverse_dict_and_create_metadata_inputs(k):
+                traverse_dict_and_create_metadata_inputs(v)
+                for k, v in d.items()
+            }
+
+        elif isinstance(d, list):
+            return [traverse_dict_and_create_metadata_inputs(el) for el in d]
+        else:
+            return d
```

Comment:
Not a blocker for this PR, as I still don't quite follow the practical usage of an importer node when pipeline accepts artifact inputs. But I think there's a bug here in case `isinstance(d, str)`, users may embed a `PipelineChannel` into a string, e.g.: `f'{task1.output}-{name}'`. and I think we are missing that parse-and-find logic here.

Reply:
> Not a blocker for this PR, as I still don't quite follow the practical usage of an importer node when pipeline accepts artifact inputs.

Agreed -- let's discuss offline about where we can raise this consideration.

> I think we are missing that parse-and-find logic here

Updated